src/core/utils/CorePlugin.ts

```diff
@@ -33,6 +35,15 @@ export class CorePlugin extends JWPlugin {
     // Public
     //--------------------------------------------------------------------------
 
+    /**
+     * Change the type of the selection. If there is no selection, change the
+     * type of the range nodes' parent.
+     *
+     * @param intent
+     */
+    formatParagraph(intent: FormatParagraphIntent): void {
+        this.editor.vDocument.formatParagraph(intent.payload.value);
```

Comment:
Why on `VDocument` ?

Reply:
That was the design as discussed: CorePlugin makes the link with VDocument - VDocument handles the changes on itself. I'm aware this will change with the commands redesign and the probable elimination of CorePlugin but in the meantime this has been the design.

Reply:
Well yes, for very "basic" operations like `insert`, but even `deleteForward`/`deleteBackward` are in `CorePlugin` so why would this one be in `VDocument` ? (To be 100% fair, I just realized I merged `applyFormat` in `VDocument`, which is wrong as well IMO since it's exactly the same use case as this one.)

Reply:
To me putting the delete functions in CorePlugin was precisely the mistake. Since it was done while I was on vacation I assumed that was just an oversight 😉 But anyways with the rethinking of Intents and Commands, I believe that pattern will change and maybe CorePlugin will even disappear...

Reply:
Well, I think we changed our minds a hundred times about CorePlugin so I can't really blame anyone for this. However, the commands PR will have no impact on this issue. This is more of a "separation of concerns" issue IMO. Like formatting in bold might only be the concern of the "Bold" plugin, and changing the paragraph might be the concern of the "Paragraph" plugin. I think we need to discuss this sooner rather than later lest we introduce more confusion on the matter.